services/src/main/java/org/keycloak/protocol/oidc/grants/ciba/endpoints/BackchannelAuthenticationEndpoint.java

```diff
@@ -0,0 +1,238 @@
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.protocol.oidc.grants.ciba.endpoints;
+
+import static org.keycloak.protocol.oidc.OIDCLoginProtocol.ID_TOKEN_HINT;
+import static org.keycloak.protocol.oidc.OIDCLoginProtocol.LOGIN_HINT_PARAM;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.POST;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.MultivaluedMap;
+import javax.ws.rs.core.Response;
+import java.util.Collections;
+import java.util.Optional;
+
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.jboss.resteasy.annotations.cache.NoCache;
+import org.jboss.resteasy.spi.HttpRequest;
+import org.keycloak.OAuth2Constants;
+import org.keycloak.OAuthErrorException;
+import org.keycloak.common.Profile;
+import org.keycloak.common.util.Time;
+import org.keycloak.events.EventBuilder;
+import org.keycloak.events.EventType;
+import org.keycloak.models.CibaConfig;
+import org.keycloak.models.ClientModel;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.OAuth2DeviceCodeModel;
+import org.keycloak.models.OAuth2DeviceTokenStoreProvider;
+import org.keycloak.models.OAuth2DeviceUserCodeModel;
+import org.keycloak.models.RealmModel;
+import org.keycloak.models.UserModel;
+import org.keycloak.protocol.oidc.grants.ciba.CibaGrantType;
+import org.keycloak.protocol.oidc.grants.ciba.channel.AuthenticationChannelProvider;
+import org.keycloak.protocol.oidc.grants.ciba.channel.AuthenticationRequest;
+import org.keycloak.protocol.oidc.grants.ciba.resolvers.CIBALoginUserResolver;
+import org.keycloak.services.ErrorResponseException;
+import org.keycloak.util.JsonSerialization;
+import org.keycloak.utils.ProfileHelper;
+
+public class BackchannelAuthenticationEndpoint extends AbstractCibaEndpoint {
+
+    private final RealmModel realm;
+
+    public BackchannelAuthenticationEndpoint(KeycloakSession session, EventBuilder event) {
+        super(session, event);
+        this.realm = session.getContext().getRealm();
+        event.event(EventType.LOGIN);
+    }
+
+    @POST
+    @NoCache
+    @Consumes(MediaType.APPLICATION_FORM_URLENCODED)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response processGrantRequest(@Context HttpRequest httpRequest) {
+        ProfileHelper.requireFeature(Profile.Feature.CIBA);
```

Comment:
Please, look at my changes here https://github.com/pedroigor/keycloak/commit/36d4dad845c47e74edd251b13f9f80034872c9c9#diff-02e6e5ccb2fbf550ae191261cd4daa7e0618fc5e52f3810ea7e69550d0ac3d3fL72.

We should avoid using this kind of check and avoid loading any CIBA-related provider by https://github.com/pedroigor/keycloak/commit/36d4dad845c47e74edd251b13f9f80034872c9c9#diff-b3048d7253e9d6afa4a39616a9e090326ef3474fc31322f2a797991eeba73a5dR35.

Here we are using `EnvironmentDependentProviderFactory` to decide whether or not CIBA providers should be loaded.

Reply:
@pedroigor I see, I would like to add them. By the way, your commit(https://github.com/pedroigor/keycloak/commit/36d4dad845c47e74edd251b13f9f80034872c9c9) have other different parts against my latest commit(https://github.com/keycloak/keycloak/pull/7679/commits/3ab5bae6a456ba9eb91cc51ab98b0e3f79feebcf). Should I cherry pick your this commit to my latest commit?

Reply:
Yes, I would say so. There is an important fix to `KeycloakSessionFactory` here too. See https://github.com/pedroigor/keycloak/commit/36d4dad845c47e74edd251b13f9f80034872c9c9#diff-84a206194f2fcaa58d8b4cc343e9092f5c7e7d22576ec362d87fa85cbfc823c9.

Reply:
@pedroigor I've cherry-picked your commit (https://github.com/pedroigor/keycloak/commit/36d4dad845c47e74edd251b13f9f80034872c9c9). Thank you. However, mentioned below, `CIBATest` fails in `-Pauth-server-wildfly`.